### PR TITLE
#930 The WebSocket is in an invalid state ('Aborted') for this operation #1030

### PR DESCRIPTION
**Pull request by OnSive**
Reopened PR: #1030 

[tdejong-tools4ever](https://github.com/tdejong-tools4ever) commented on [Sep 30, 2019](1030#issue-500284522)

### CODE REVIEW

File: src/Ocelot/WebSockets/Middleware/WebSocketsProxyMiddleware.cs [hunk 1]
```diff
@@ -48,13 +48,21 @@ private static async Task PumpWebSocket(WebSocket source, WebSocket destination,
                 catch (OperationCanceledException)
                 {
                     await destination.CloseOutputAsync(WebSocketCloseStatus.EndpointUnavailable, null, cancellationToken);
+                    if (destination.State == WebSocketState.Open || destination.State == WebSocketState.CloseReceived)
+                    {
+                        await destination.CloseOutputAsync(WebSocketCloseStatus.EndpointUnavailable, null, cancellationToken);
+                    }
```
Comment (raman-m) [inline src/Ocelot/WebSockets/Middleware/WebSocketsProxyMiddleware.cs:54]: Simple logical challenge:
How many times will the action be applied here?
Once or twice?

The action is `CloseOutputAsync`.

File: src/Ocelot/WebSockets/Middleware/WebSocketsProxyMiddleware.cs [hunk 2]
```diff
@@ -48,13 +48,21 @@ private static async Task PumpWebSocket(WebSocket source, WebSocket destination,
...
                 }
                 catch (WebSocketException e)
                 {
                     if (e.WebSocketErrorCode == WebSocketError.ConnectionClosedPrematurely)
                     {
                         await destination.CloseOutputAsync(WebSocketCloseStatus.EndpointUnavailable, null, cancellationToken);
+                        if (destination.State == WebSocketState.Open || destination.State == WebSocketState.CloseReceived)
+                        {
+                            await destination.CloseOutputAsync(WebSocketCloseStatus.EndpointUnavailable, null, cancellationToken);
+                        }
```
Comment (raman-m) [inline src/Ocelot/WebSockets/Middleware/WebSocketsProxyMiddleware.cs:65]: Why do you apply the same action twice?

The action is `CloseOutputAsync`.

**Review by raman-m — CHANGES_REQUESTED** [review]
I do NOT see any fixes!
I see extra code only with extra post-action.

**Comment by OnSive** [comment]
I'm closing this PR as I can't remember what it was actually about and the original author is actively communicating on #1030

**Pull request closed without merging**